haystack_experimental/components/generators/chat/openai.py

```diff
@@ -163,6 +163,8 @@ def __init__(  # noqa: PLR0913
             Whether to enable strict schema adherence for tool calls. If set to `True`, the model will follow exactly
             the schema provided in the `parameters` field of the tool definition, but this may increase latency.
         """
+        if tools and len(tools) != len({tool.name for tool in tools}):
+            raise ValueError("Duplicate tool names are not allowed.")
```

Comment:
Let's update the error message to display the set of non-unique names.

Reply:
done